### ISSUE TITLE
Function "Warmwasser Push"

### ISSUE THREAD

**Issue by mathiaslehmann** [erikkastelec/hass-WEM-Portal#23]
Hello, I have a question about the request at the time for the hot water push function. Where can I find this function in the current version. I refer to #7 

Thank you very much

**Comment by erikkastelec** [comment]
After updating the integration you should see the entity under entities in Home Assistant.
If you go to https://your_homeassistant_ip/config/entities and search for wemportal you should get all the available entities for this integration. All the entities of type number of select are used for changing settings. 

For me  WW-Push is named nachladung inside Home Assistant (select.nachladung).
I hope this helps.

**Comment by mathiaslehmann** [comment]
Hi, thanks for the quick response. I had already discovered the new entities, but unfortunately the entity "select.nachladung" is missing. I only have the entities "select.betriebsart", "select.party_heizpause_funktion", "select.party_heizpause_raumsoll_temperatur", "select.urlaub_funktion" and "select.urlaub_raumsoll_temperatur"

**Comment by erikkastelec** [comment]
Do you have hot water push functionality inside WEM mobile app?

**Comment by mathiaslehmann** [comment]
Yes

**Comment by mathiaslehmann** [comment]
![Screenshot_20220420-142315_WEM](https://user-images.githubusercontent.com/16978467/164230480-a2b86abe-8a08-4bb2-a141-09fc7995ee29.jpg)

**Comment by erikkastelec** [comment]
Ok, I think I see the issue. I only added settings where you select from multiple options or enter a number eg. set temperature. This setting seems to be on/off so I left it out.

Thanks for pointing this out, will try to fix this as soon as possible.

**Comment by mathiaslehmann** [comment]
Yes, I can only press the "Warmwasser Push" button and then I get the question whether the function should be activated. See the screenshot. Thank you for the amazing support.

![Screenshot_20220420-143634_WEM](https://user-images.githubusercontent.com/16978467/164232941-e712f366-a357-4941-b1ef-d662e4a1a013.jpg)
.

**Comment by iridium2001** [comment]
@erikkastelec: if I recall correctly, Hot Water Push also is a numeric value, which the API seems to interpret as a boolean. So this can be set to e.g. 1 to activate. I think I included this in my first posts on this topic in the other thread.

**Comment by erikkastelec** [comment]
I used your comment from a previous issue for reference. The type is set to 2, which is weird as for me this denotes a schedule(holidays...), so I made a special case for this.
 Just needs a few fixes, so expect the change to be released next week, as I am away for the weekend

**Comment by iridium2001** [comment]
@erikkastelec: 

Not really sure what you mean by "type 2" but here are the links to my previous post on this topic:
https://github.com/erikkastelec/hass-WEM-Portal/issues/9#issuecomment-1036601489
and 
https://github.com/erikkastelec/hass-WEM-Portal/issues/9#issuecomment-1039650494

**Comment by iridium2001** [comment]
Hi Erik, just wanted to ask if there is still s chance that we can get this function. For me personally, this is one of the most interesting parameters to control remotely from home assistant. I plan to use it to reduce water temperature while we are out of the house and sail get convenient temperature automatically upon return.

**Comment by erikkastelec** [comment]
I will try to get it done this week. Had to complete some university assignments and forgot about this.

**Comment by erikkastelec** [comment]
Expect this between 6 and 12 June, as I am currently preoccupied with exams.

**Comment by mathiaslehmann** [comment]
Hello and thanks for the work. Maybe I'll come and test it tonight. To install the module from the dev branch I just have to change the contents of the folder "wemportal" under "custom_components", right?

**Comment by erikkastelec** [comment]
Yes, that's it.

**Comment by mathiaslehmann** [comment]
Hello, unfortunately my health has failed and I am not able to test.  Maybe @iridium2001 can test it.

**Comment by iridium2001** [comment]
Thanks Eric for implementing this! Will try to test it tonight. Already installed the dev version.

One question: the button will not be reset automatically, correct?

Best regards, Ingo

09.06.2022 19:31:02 mathiaslehmann ***@***.***>:

**Comment by iridium2001** [comment]
Warmwasser-Push works fine for me. 
The state can be manually reset and the system will continue to heat up the water until the correct temperature is reached. For the reset of the button state a small automation in HA will do the job. 
Perhaps Wemportal is also resetting the state once the heat up cycle is completed, but for testing this I first need to wait until my water has cooled down. :-) 
Will let you know tomorrow.
Thanks again @erikkastelec for implementing this!

**Comment by iridium2001** [comment]
I can confirm that the switch state is automatically reset to "false" when the push function reached the target temperature. No need to artificially reset it in HA.

Testing was a bit compliated. When I set a new (higher) target temparature, the heating immediately starts warming up the water.  And then the push function would not be possible for several hours. The only way I found was to change the water temperature schedule directly after a new temperature was configured (and define the current time as a "low" phase).
